### ISSUE TITLE
RStudio Server 2022.07.0+548 does not seem to work on mybinder.org

### ISSUE THREAD

**Issue by eitsupi** [rocker-org/rocker-versioned2#505]
### Container image name

rocker/binder:4.2.1

### Container image digest

rocker/binder@sha256:5166277a9d98da6e136d502dd63b1e8377276444efef57767eedf87928b9bcce

### What operating system are you seeing the problem on?

Linux

### System information

mybinder.org

### Bug description

It appears that `rocker/binder` with the latest RStudio Server installed cannot use RStudio on Binder.
It works fine locally.

cc @yuvipanda

### How to reproduce this bug?

```sh
Open https://mybinder.org/v2/gh/rocker-org/binder/321a01a592b71d4d0e237206755e94f557139f1e and launch RStudio.
```

**Comment by eitsupi** [comment]
This problem still does not seem to have been fixed. (#601)

@cboettig @yuvipanda I thought rocker/binder was confusing and should be discontinued. Now there is another recommended way to use R in the binder, right?
https://mybinder.readthedocs.io/en/latest/using/config_files.html#runtime-txt

**Comment by cboettig** [comment]
@eitsupi I think this is worth revisiting.  binder has always had this non-rocker-based mechanism for R/RStudio, but it is still based on Ubuntu 18.04 as the base image, and it looks like the example still serves R 4.1.3, http://mybinder.org/v2/gh/binder-examples/r/master?urlpath=rstudio.  I don't think that situation is ideal.  

Being able to layer the binder app transparently on top of the existing rocker-versioned image seems like it would be preferable, but I'm not up-to-speed on the compatibility issues between the latest RStudio versions and the latest RStudio-jupyter integration.  @yuvipanda or @ryanlovett would know best.

**Comment by eitsupi** [comment]
@cboettig Thank you for sharing your thoughts.

> but it is still based on Ubuntu 18.04 as the base image, and it looks like the example still serves R 4.1.3, http://mybinder.org/v2/gh/binder-examples/r/master?urlpath=rstudio. I don't think that situation is ideal.

I think it is just that the example has not been updated and we can actually use the latest version of R. For example, the following repository.
https://github.com/uribo/cue2022aw_r104/blob/ed4d5a935b84958f4375850026eb09d1090fbf3f/runtime.txt#L1

**Comment by cboettig** [comment]
right, the R version is configurable but it still builds on 18.04 at the moment.  Certainly in many cases that is unlikely to create any issues, but users will have to know which apt packages to add to apt.txt and may be surprised when behavior of some packages is quite different (osgeo libraries for instance).

Perhaps the ssl error mentioned above is also a result of the newer ssl libraries on more recent ubuntu releases?  I was only thinking that if this was an issue that could be resolved upstream in  jupyter-rsession-proxy users would not have to navigate such edge cases.

**Comment by eitsupi** [comment]
> right, the R version is configurable but it still builds on 18.04 at the moment. Certainly in many cases that is unlikely to create any issues, but users will have to know which apt packages to add to apt.txt and may be surprised when behavior of some packages is quite different (osgeo libraries for instance).

That makes sense.

> Perhaps the ssl error mentioned above is also a result of the newer ssl libraries on more recent ubuntu releases? I was only thinking that if this was an issue that could be resolved upstream in jupyter-rsession-proxy users would not have to navigate such edge cases.

I don't think it is a matter of difference between OpenSSL 1 and 3 since both `rocker/binder:4.2.0` and `rocker/binder:4.2.1` are based of `ubuntu:focal`.

However, it seems that RStudio Server 2022.07 started supporting `ubuntu:jammy`, so it may have something to do with it.

**Comment by yuvipanda** [comment]
https://github.com/jupyterhub/repo2docker/pull/909 deals with openssl on repo2docker, but it isn't merged yet.

I started working on documentation for how to build images for JupyterHub where R is the primary use case, and recommend basing it off rocker: https://jupyterhub-image.guide/rocker.html
